### PR TITLE
add test to check for support for obsoleted curves

### LINKED ISSUE #539: Obsolete curves in TLS 1.3 ClientHello
# New test script idea

## What TLS message this idea relates to?

ClientHello

## What TLS extension this idea relates to?

`supported_groups`, `key_share`

## What is the behaviour the test script should test?

* ClientHello that includes TLS 1.3 and only a group that is marked as obsolete in `supported_groups`  is rejected with `illegal_parameter`
* TLS 1.3 ClientHello that includes both a valid group and an obsolete group in `supported_groups` and a key share for the valid group is rejected with `illegal_parameter`
* TLS 1.3 ClientHello that includes a valid group and a key share for a obsolete group is rejected with `illegal_parameter` (inconsistent key_share)
* TLS 1.3 ClientHello that includes an obsolete group and a key share for obsolete group is rejected with `illegal_parameter`
* the same happens for a resuming (PSK) handshake

## Are there scripts that test related functionality?

`test-tls13-unrecognised-groups.py` but it's for groups from outside defined range

## Additional information

RFC 8446, Section B.3.1.4.

### PR DESCRIPTION
**Pull request by tomato42**
### Description

Test coverage for a MUST requirement from TLS 1.3

### Motivation and Context

fixes #539 
blocked by https://github.com/tomato42/tlslite-ng/issues/389

### Checklist

- [x] I have read the [CONTRIBUTING.md](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md) document and my PR follows [change requirements](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md#change-requirements) therein
- [x] the changes are also reflected in documentation and code comments
- [x] all new and existing tests pass (see Travis CI results)
- [x] [test script checklist](https://github.com/tomato42/tlsfuzzer/wiki/Test-script-checklist) was followed for new scripts
- [x] new test script added to `tlslite-ng.json` and `tlslite-ng-random-subset.json`
- [x] new and modified scripts were ran against popular TLS implementations:
  - [x] OpenSSL https://github.com/openssl/openssl/issues/8369
  - [x] NSS [MZBZ#1554760](https://bugzilla.mozilla.org/show_bug.cgi?id=1554760)
  - [x] GnuTLS [gnutls/gnutls#908](https://gitlab.com/gnutls/gnutls/issues/908)
- [x] required version of tlslite-ng updated in requirements.txt and README.md
- [x] sanity check that HRR will be employed by server for empty key_share ext

---
This change is [Reviewable](https://reviewable.io/reviews/tomato42/tlsfuzzer/540)

### CODE REVIEW

File: scripts/test-tls13-obsolete-curves.py [hunk 1]
```diff
@@ -155,17 +157,181 @@ def main():
...
+            .create(SIG_ALL)
         node = node.add_child(ClientHelloGenerator(ciphers, extensions=ext))
         node = node.add_child(ExpectAlert(AlertLevel.fatal,
                                           alert_desc))
 
         node = node.add_child(ExpectClose())
-        conversation_name = "{0} in TLS 1.3".format(obsolete_group_name)
+        conversation_name = \
+            "{0} in key_share and secp256r1 in "\
+            "supported_groups (inconsistent extensions)"\
```
Comment (ueno) [inline scripts/test-tls13-obsolete-curves.py]: given those new tests end up with the same consequence (receive alert + close connection), wouldn't it be possible to reuse a single test function with different configurations?

Comment (tomato42) [reply scripts/test-tls13-obsolete-curves.py]: there are slight differences for expected alert, but yes, except for key_share and supported_groups the extensions are the same for all connections. Will look into fixing it.

File: scripts/test-tls13-obsolete-curves.py [hunk 2]
```diff
@@ -42,6 +42,9 @@ def help_msg():
     print("                may be specified multiple times")
     print(" -n num         only run `num` random tests instead of a full set")
     print("                (\"sanity\" tests are always executed)")
+    print(" -a alert_desc  Alert description expected for cases with invalid")
+    print("                curves advertised, integer of name.")
```
[the remark this answers is not in the document]
Comment (tomato42) [reply scripts/test-tls13-obsolete-curves.py]: fixed

**Comment by tomato42** [comment]
@ueno ok, should be fixed, please take a second look

**Pull request merged**